src/generate/rust.rs

```diff
@@ -459,12 +485,27 @@ impl Continuation<'_> {
 
 trait ContFn {
     fn apply(self, cont: Continuation<'_>) -> Continuation<'_>;
+    // HACK(eddyb) `Box<dyn FnOnce<A>>: FnOnce<A>` is not stable yet,
+    // so this is needed to implement `ContFn` for `Box<dyn ContFn>`.
+    fn apply_box(self: Box<Self>, cont: Continuation<'_>) -> Continuation<'_>;
 }
 
 impl<F: FnOnce(Continuation<'_>) -> Continuation<'_>> ContFn for F {
     fn apply(self, cont: Continuation<'_>) -> Continuation<'_> {
         self(cont)
     }
+    fn apply_box(self: Box<Self>, cont: Continuation<'_>) -> Continuation<'_> {
+        (*self).apply(cont)
+    }
+}
+
+impl<'a> ContFn for Box<dyn ContFn + 'a> {
```

Comment:
You should be able to elide the lifetime in the header here?

Reply:
Argh, I forgot about that! I think I did use `'_` the first attempt.